hbase-server/src/main/java/org/apache/hadoop/hbase/mob/MobFileCleanerChore.java

```diff
@@ -0,0 +1,331 @@
+      MobConstants.MOB_CLEANER_PERIOD);
+  }
+
+  public MobFileCleanerChore(HMaster master) {
+    super(master.getServerName() + "-ExpiredMobFileCleanerChore", master,
+        master.getConfiguration().getInt(MobConstants.MOB_CLEANER_PERIOD,
+          MobConstants.DEFAULT_MOB_CLEANER_PERIOD),
+        master.getConfiguration().getInt(MobConstants.MOB_CLEANER_PERIOD,
+          MobConstants.DEFAULT_MOB_CLEANER_PERIOD),
+        TimeUnit.SECONDS);
+    this.master = master;
+    cleaner = new ExpiredMobFileCleaner();
+    cleaner.setConf(master.getConfiguration());
+    checkObsoleteConfigurations();
+  }
+
+  private void checkObsoleteConfigurations() {
+    Configuration conf = master.getConfiguration();
+
+    if (conf.get("hbase.mob.compaction.mergeable.threshold") != null) {
+      LOG.warn("'hbase.mob.compaction.mergeable.threshold' is obsolete and not used anymore.");
+    }
+    if (conf.get("hbase.mob.delfile.max.count") != null) {
+      LOG.warn("'hbase.mob.delfile.max.count' is obsolete and not used anymore.");
+    }
+    if (conf.get("hbase.mob.compaction.threads.max") != null) {
+      LOG.warn("'hbase.mob.compaction.threads.max' is obsolete and not used anymore.");
+    }
+    if (conf.get("hbase.mob.compaction.batch.size") != null) {
+      LOG.warn("'hbase.mob.compaction.batch.size' is obsolete and not used anymore.");
+    }
+  }
+
+  @VisibleForTesting
+  public MobFileCleanerChore() {
+    this.master = null;
+  }
+
+  @Override
+  @edu.umd.cs.findbugs.annotations.SuppressWarnings(value = "REC_CATCH_EXCEPTION",
+      justification = "Intentional")
+
+  protected void chore() {
+    TableDescriptors htds = master.getTableDescriptors();
+
+    Map<String, TableDescriptor> map = null;
+    try {
+      map = htds.getAll();
+    } catch (IOException e) {
+      LOG.error("MobFileCleanerChore failed", e);
+      return;
+    }
+    for (TableDescriptor htd : map.values()) {
+      for (ColumnFamilyDescriptor hcd : htd.getColumnFamilies()) {
+        if (hcd.isMobEnabled() && hcd.getMinVersions() == 0) {
+          try {
+            cleaner.cleanExpiredMobFiles(htd.getTableName().getNameAsString(), hcd);
+          } catch (IOException e) {
+            String errMsg = String.format("Failed to clean the expired mob files table=%s" +
+                " family=%s", htd.getTableName().getNameAsString(), hcd.getNameAsString());
+            LOG.error(errMsg, e);
+          }
+        }
+      }
+      try {
+        // Now clean obsolete files for a table
+        LOG.info("Cleaning obsolete MOB files from table={}", htd.getTableName());
+        cleanupObsoleteMobFiles(master.getConfiguration(), htd.getTableName());
+        LOG.info("Cleaning obsolete MOB files finished for table={}", htd.getTableName());
+      } catch (IOException e) {
+        String errMsg =
+            String.format("Failed to clean the obsolete mob files for table=",htd.getTableName());
+        LOG.error(errMsg, e);
```

Comment:
What method can I use with parametrized message and throwable object?

Reply:
the slf4j Logger methods for messages take vararg to do matching to parameters and also have special handling so that the very last parameter can be an exception.

ref: http://www.slf4j.org/faq.html#paramException